code/config/lib/Loot/Config/Yaml.hs

```diff
@@ -95,3 +96,71 @@ parseMulti p = keyOrDefault (fromString $ symbolVal p) mempty configParser
 
 instance OptionsFromJson is => FromJSON (ConfigRec 'Partial is) where
     parseJSON = toAesonParser' configParser
+
+
+-- | This class is helper which converts config to object
+class OptionsToJson (is :: [ItemKind]) where
+  -- | Convert Config to Object
+  configToObject :: ConfigRec 'Final is -> Object
+
+instance OptionsToJson '[] where
+  configToObject RNil = HM.empty
+
+instance
+  forall l v is
+  . ( KnownSymbol l
+    , ToJSON v
+    , OptionsToJson is
+    )
+  => OptionsToJson ((l ::: v) ': is) where
+  configToObject (ItemOptionF x :& xs) =
+    HM.insert key value $ configToObject xs
+    where
+      key = fromString $ symbolVal (Proxy :: Proxy l)
+      value = toJSON x
+
+instance
+  forall l us is
+  . ( KnownSymbol l
+    , Monoid (ConfigRec 'Partial us)
+    , OptionsToJson us
+    , OptionsToJson is
+    )
+  => OptionsToJson ((l ::< us) ': is) where
+  configToObject (ItemSub x :& xs) =
+    HM.insert key value $ configToObject xs
+    where
+      key = fromString $ symbolVal (Proxy :: Proxy l)
+      value = toJSON x
+
+instance
+  forall l us is
+  . ( KnownSymbol l
+    , Monoid (ConfigRec 'Partial (SumSelection l : us))
+    , OptionsToJson (SumSelection l : us)
+    , OptionsToJson is
+    )
+  => OptionsToJson ((l ::+ us) ': is) where
+  configToObject (ItemSumF x :& xs) =
+    HM.insert key value $ configToObject xs
+    where
+      key = fromString $ symbolVal (Proxy :: Proxy l)
```

Comment:
As Pasquale pointed out in another PR, this `fromString $ symbolVal (Proxy :: Proxy l)` thing is used in 4 places, so it makes sense to extract it into a function. This function can take `l` type via type applications instead of `Proxy` (that seems neater, but it's up to you).

Reply:
I thought about this duplications
will do it